docs/current_docs/manuals/developer/functions.mdx

```diff
@@ -338,6 +338,109 @@ The result will look like this:
 Hello, world
 ```
 
+## Contextual values
+
+Function arguments can define a default contextual value if no value is supplied for them. This default value will be load prior to the function call which makes it more cache efficient.
+
+:::note
+Contextual values are only available for arguments of type `Directory` and `File` for now.
+
+It's specially useful for static directory or file provided to the module. For example, a codebase source's code.
+:::
+
+The contextual value consist of a path to a directory or a file set through the `defaultPath` parameter metadata pointing to a location inside the module's context directory.
+- If the path is absolute, its root is the path to the git repository containting that module (called `context directory`), falling into the location of the `dagger.json` if no `.git` directory is found.
```

Comment:
contain~t~ing

Just thought I'd point it out while I was reading how to use them, thank you for writing them up :100:

Reply:
Thanks for catching that!